### PR TITLE
Added SD card support option for extraction of OBB file.

### PR DESCRIPTION
**Pull request by sivakumarpaidi**

### CODE REVIEW

**Comment by sivakumarpaidi** [comment]
|PR checklist|Status|
|-----|:------:|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA|
|Is code commenting done properly for the changes made?|No|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|No|
|Are the message strings used in the application taken from strings.xml file?|Yes|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|NA|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|NA|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|No|

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/DownloadExpansionFile.java [hunk 1]
```diff
@@ -56,13 +56,7 @@
...
                     9, // the version of the APK that the file was uploaded
                     // against
-                    1989909475L // the length of the file in bytes
-            ),
-            new XAPKFile(
-                    false, // false signifies a patch file
-                    12, // the version of the APK that the file was uploaded
-                    // against
-                    188120L // the length of the file in bytes
+                    1797373003L // the length of the file in bytes
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/DownloadExpansionFile.java]: Revert this change

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java [hunk 2]
```diff
@@ -125,29 +128,43 @@ public static OBGroup armPointer() {
...
 
         // Retrieve the stored values of main and patch file version
-        int storedMainFileVersion = sharedPref.getInt(getString(R.string.mainFileVersion), defaultFileVersion);
-        int storedPatchFileVersion = sharedPref.getInt(getString(R.string.patchFileVersion), defaultFileVersion);
-        boolean isExtractionRequired = isExpansionExtractionRequired(storedMainFileVersion, storedPatchFileVersion);
-
-        if (storedMainFileVersion == 0 && storedPatchFileVersion == 0) {
-            // Set main and patch file version to 0, if the extractions takes place for the first time
+        int storedMainFileVersion = sharedPref.getInt("mainFileVersion", defaultFileVersion);
+        int storedPatchFileVersion = sharedPref.getInt("patchFileVersion", defaultFileVersion);
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java]: Use getString() method, as it was being used earlier

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java [hunk 3]
```diff
@@ -104,6 +126,83 @@ private boolean isExpansionExtractionRequired(int storedMainFileVersion, int sto
...
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.RECORD_AUDIO}, 1);
+        } else {
+            new DownloadFile().execute(null, null, null);
+        }
+    }
+
+    public String getDataFilePath() {
+        String internalDataFilePath = null;
+        String externalDataFilePath = null;
+        String DataFilePath = null;
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java]: Use camel case

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java [hunk 4]
```diff
@@ -104,6 +126,83 @@ private boolean isExpansionExtractionRequired(int storedMainFileVersion, int sto
...
+//              For internal storage path
+                internalDataFilePath = file.getAbsolutePath() + File.separator;
+            }
+        }
+        if (externalDataFilePath == null) {
+            DataFilePath = internalDataFilePath;
+        } else if (sharedPref.getInt(getString(R.string.dataPath), 0) == 2) {
+            DataFilePath = externalDataFilePath;
+        }
+        Assetspath = DataFilePath;
```
Comment (akashdabhi03) [inline app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java]: Camel case

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/DownloadExpansionFile.java (hunk 1 above)
Comment (sivakumarpaidi) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/DownloadExpansionFile.java]: Done

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java (hunk 2 above)
Comment (sivakumarpaidi) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java]: Done

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java (hunk 4 above)
Comment (sivakumarpaidi) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java]: Done.

File: app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java (hunk 3 above)
Comment (sivakumarpaidi) [reply app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/SplashScreenActivity.java]: Done.

**Review by akashdabhi03 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]
